src/fs.cc

```diff
@@ -6,12 +6,18 @@
 #include <unistd.h>
 #include <cerrno>
 #include <ftw.h>
+#include <string.h>
 
 #include <string>
 #include <vector>
 #include <fstream>
 #include <sstream>
 #include <streambuf>
+#include <iomanip>
+#include <stack>
+#include <algorithm>
+
+#include <openssl/sha.h>
```

Comment:
i suppose this resolves the long-time issue of OpenSSL vs GNUTLS compilation WRT cURL.

Reply:
Ideally, I'd use IFDEF to use either, but I've never used GNUTLS before.

Reply:
I personally don't care. There's always the option of implementing sha256 :tada:.

But, you know, that's not really worthwhile.